src/DetailsView/components/switcher.tsx

```diff
@@ -27,6 +28,12 @@ export class Switcher extends React.Component<SwitcherProps, SwitcherState> {
         this.state = { selectedKey: props.pivotKey };
     }
 
+    public componentDidUpdate(prevProps: Readonly<SwitcherProps>): void {
+        if (isEqual(prevProps, this.props) === false) {
```

Comment:
nit: do we want to check for all the props and perform a deep comparison here or can it be improved by comparing some specific prop or property?